node-cli/users.js

```diff
@@ -0,0 +1,106 @@
+const inquirer = require("inquirer");
+const Realm = require("realm");
+const index = require("./index");
+const config = require("./config");
+const main = require("./main");
+const output = require("./output");
+
+const REALM_APP_ID = config.realmAppId;
+const appConfig = {
+  id: REALM_APP_ID,
+  timeout: 10000,
+};
+
+const app = new Realm.App(appConfig);
+
+authedUser = {};
+
+async function logIn() {
+  const input = await inquirer.prompt([
+    {
+      type: "input",
+      name: "email",
+      message: "Email:",
+    },
+    {
+      type: "password",
+      name: "password",
+      message: "Password:",
+      mask: "*",
+    },
+  ]);
+
+  try {
+    const credentials = Realm.Credentials.emailPassword(
+      input.email,
+      input.password
+    );
+    const user = await app.logIn(credentials);
```

Comment:
If `app.logIn()` resolves then it was successful, otherwise it throws. We should handle login errors in the catch block.

```js
try {
  const credentials = Realm.Credentials.emailPassword(
    input.email,
    input.password
  );
  await app.logIn(credentials);
  output.result("You have successfully logged in as " + app.currentUser.id);
  return main.mainMenu();
} catch (err) {
  output.error("There was an error logging you in");
  output.error(JSON.stringify(err, " ", 3));
  return logIn();
}
```